gsy_framework/sim_results/electric_blue/volume_timeseries.py

```diff
@@ -0,0 +1,160 @@
+from typing import Dict, Optional
+
+from pendulum import DateTime
+
+from gsy_framework.enums import AggregationResolution, AvailableMarketTypes
+from gsy_framework.forward_markets.aggregated_ssp import get_aggregated_SSP
+from gsy_framework.forward_markets.forward_profile import (
+    ForwardTradeProfileGenerator)
+from gsy_framework.sim_results.electric_blue.aggregate_results import (
+    ForwardDeviceStats)
+
+FORWARD_MARKET_TYPES = [
+    AvailableMarketTypes.YEAR_FORWARD, AvailableMarketTypes.MONTH_FORWARD,
+    AvailableMarketTypes.WEEK_FORWARD, AvailableMarketTypes.DAY_FORWARD,
+]
+
+# a dictionary showing the start of time slot for each resolution.
+START_OF = {
+    AggregationResolution.RES_1_HOUR: "hour",
+    AggregationResolution.RES_1_WEEK: "week",
+    AggregationResolution.RES_1_MONTH: "month",
+    AggregationResolution.RES_1_YEAR: "year",
+}
+
+
+class AssetVolumeTimeSeries:
+    """This class generates combined volume time series for the whole year for each asset.
+    The result would be like this for a monthly resolution for the year 2020:
+    {
+        DateTime(2020, 1, 1, 0, 0, 0): {
+            DateTime(2020, 1, 1, 0, 0, 0): {
+                "SSP": 1005.4160297565659,
+                "YEAR_FORWARD": {
+                    "bought_kWh": 201.08320595131318,
+                    "sold_kWh": 201.08320595131318,
+                },
+                "MONTH_FORWARD": {
+                    "bought_kWh": 201.08320595131318,
+                    "sold_kWh": 201.08320595131318,
+                },
+                "WEEK_FORWARD": {
+                    "bought_kWh": 32.432775153437625,
+                    "sold_kWh": 32.432775153437625,
+                },
+                "DAY_FORWARD": {
+                    "bought_kWh": 6.486555030687522,
+                    "sold_kWh": 6.486555030687522,
+                },
+            },
+            DateTime(2020, 2, 1, 0, 0, 0): {...},
+            ...
+            DateTime(2020, 12, 1, 0, 0, 0): {...},
+        },
+    }
+    """
+    def __init__(
+            self, asset_uuid: str, asset_peak_kWh: float,
+            resolution: AggregationResolution):
+
+        self.asset_uuid = asset_uuid
+        self.asset_peak_kWh = asset_peak_kWh
+        self.resolution = resolution
```

Comment:
So there will be one `AssetVolumeTimeSeries` per resolution? So I understand that not all resolutions have to be updated every time a new trade has happened, right? So we could update the weekly resolution only if a new week has started, right?
Or are you planning to run `add()` after every trade?

Reply:
Yes. There will be one AssetVolumeTimeSeries per resolution and per asset.
No. All resolutions (of the specified device) will get updated if you receive a trade.
Also, not all years get updated when you receive a trade.